internal/persistence/postgres/pool.go

```diff
@@ -0,0 +1,154 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"go.uber.org/zap"
+)
+
+var pools = map[string]PoolOpenFunc{}
+
+// Pool is the interface of the pgxpool.Pool type, used by the postgres
+// persistence layer to query the database. A custom pool type can be
+// registered using `RegisterPool` if it implements this interface.
+type Pool interface {
```

Comment:
since `*pgxpool.Pool` is a type, I cannot use it as the abstraction type/interface for my different pool implementations. e.g. My custom aurora pool implementation is returning a `*AuroraPool`. I need a way that our postgres persistence layer can rely on and just call the query execution functions. does that make sense or I am missing something?